### ISSUE TITLE
Refresh Table Metadata doesn't work on tabular model once calculation groups are implemented

### ISSUE THREAD

**Issue by POMelisa** [otykier/TabularEditor#508]
Hi guys

So I implemented time intelligence as a calc. group in my model. Afterwards, I added 2 new columns to my Date view but when I try to refresh the table metadata to get the new columns into my model, it does not detect any changes:

![tab_editor1](https://user-images.githubusercontent.com/38052880/85856047-8bc40900-b7b7-11ea-83eb-aa4e577ec092.png)

Please help :)

/Melisa

**Comment by otykier** [comment]
I'm not able to reproduce this in the [latest version of Tabular Editor](https://github.com/otykier/TabularEditor/releases/latest). Could you please try to upgrade to see if this solves the issue in your case?

**Comment by POMelisa** [comment]
@otykier 
I just did, but it does not appear to solve the problem:

![tabed2](https://user-images.githubusercontent.com/38052880/85988047-c95ea700-b9ee-11ea-8a5e-588f35f74b33.png)

//Melisa

**Comment by otykier** [comment]
What happens when you click the button? Are you getting an error message, or does the application hang?

Also, "Refresh Table Metadata" only works when you're using provider data sources (legacy) - not structured (power query). In other words, the partition source queries should contain SQL statements, not M code. Lastly, the connection string of the data source may need to be updated to include credentials, as these are never persisted to disk, and Tabular Editor needs the credentials in order to connect to the source and check the schema.

**Comment by POMelisa** [comment]
@otykier 
Nothing of the sort, it keeps saying "No changes detected." But that makes sense now because I can see that some source queries are based on M code and some on SQL statements. I tested the feature on one table based on a SQL source query and it works fine. 

Since this dataset was built by an ex-colleague, I am not familiar with the difference between working with M code instead of SQL, and I can't find information online - does it have any repercussions if I change them from M to SQL? And more importantly, how do I do so? 

Thank you for the quick help!

**Comment by otykier** [comment]
There should be no repercussions of switching from M to SQL, provided that the M code does not apply any data transformations, and that all your data sources are relational databases that lets you use OLE DB/ODBC for querying data (SQL Server, Azure SQL DB, Oracle, etc.). Basically, if your tables can be populated using a simple `SELECT * FROM <source view/table>`, then you might as well use SQL. In order to do so, go to the Partitions below each table, right-click the "Partitions"-object and add a new partition (legacy). Then, enter the SQL statement as the partition query and then delete the original power query partition. This assumes that your model contains a Provider Data Source, which it sounds like it already does, since you were able to use the "Refresh Table Metadata" successfully on one of your tables.

I wrote a [blog post](https://tabulareditor.com/2019/07/03/TabularEditorTricks2.html) which is sort of relevant here (migrating partitions from power query to legacy).

Closing this issue, but feel free to open a new issue if you have any specific problems with adding legacy partitions to your model or getting the Refresh Table Metadata option to work with those partitions.
